### PR TITLE
Extend sync stats

### PR DESCRIPTION
**Pull request by hamza-vd**
See #895

### CODE REVIEW

File: opensrp-core/res/values/strings.xml [hunk 1]
```diff
@@ -457,4 +457,21 @@
...
+    <string name="app_info">App Info</string>
+    <string name="device_info">Device Info</string>
+    <string name="manufacturer">Manufacturer:</string>
+    <string name="device">Device:</string>
+    <string name="os_version">OS Version:</string>
+    <string name="sync_statistics">Sync Statistics</string>
+    <string name="user_info">User Info</string>
+    <string name="build_date">Build Date:</string>
+    <string name="date">Date:</string>
+    <string name="user">User:</string>
```
Comment (qiarie) [inline opensrp-core/res/values/strings.xml:476]: Do a translation of these

File: opensrp-core/src/main/java/org/smartregister/view/interactor/StatsFragmentInteractor.java [hunk 2]
```diff
@@ -108,40 +128,71 @@ public void run() {
...
+        syncInfoMap.put(USER_TEAM, sharedPreferences.fetchDefaultTeam(sharedPreferences.fetchRegisteredANM()));
+        syncInfoMap.put(USER_LOCALITY, sharedPreferences.fetchCurrentLocality());
+    }
+
+
+    private void populateBuildInfo() {
+        try {
+            syncInfoMap.put(APP_VERSION_NAME, getBuildConfigValue("VERSION_NAME"));
+            syncInfoMap.put(APP_VERSION_CODE, getBuildConfigValue("VERSION_CODE"));
+            syncInfoMap.put(DB_VERSION, getBuildConfigValue("DATABASE_VERSION"));
```
Comment (qiarie) [inline opensrp-core/src/main/java/org/smartregister/view/interactor/StatsFragmentInteractor.java:180]: See earlier comment on `DB_VERSION`

File: opensrp-core/src/test/java/org/smartregister/interactor/StatsFragmentInteractorTest.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        ReflectionHelpers.setField(presenter, "view", view);
+    }
+
+    @Test
+    public void testPopulateUserInfoPopulatesCorrectValues() {
+        StatsFragmentInteractor statsFragmentInteractor = new StatsFragmentInteractor(presenter);
+
+        CoreLibrary coreLibrary = Mockito.mock(CoreLibrary.class);
+        ReflectionHelpers.setStaticField(CoreLibrary.class, "instance", coreLibrary);
+//
```
Comment (qiarie) [inline opensrp-core/src/test/java/org/smartregister/interactor/StatsFragmentInteractorTest.java:61]: Carry out some cleanup

**Review by qiarie — CHANGES_REQUESTED** [review]

File: opensrp-core/src/main/java/org/smartregister/view/interactor/StatsFragmentInteractor.java (hunk 2 above)
Comment (hamza-vd) [reply opensrp-core/src/main/java/org/smartregister/view/interactor/StatsFragmentInteractor.java]: Refactored implementation to return a default value

File: opensrp-core/build.gradle [hunk 4]
```diff
@@ -92,7 +92,8 @@ android {
         buildConfigField "int", "DB_ENCRYPTION_VERSION", '1'
         buildConfigField("int", "VERSION_CODE", "${defaultConfig.versionCode}")
         buildConfigField("String", "VERSION_NAME", "\"${defaultConfig.versionName}\"")
-    } 
+        buildConfigField "int", "DATABASE_VERSION", '1'
```
[the remark this answers is not in the document]
Comment (hamza-vd) [reply opensrp-core/build.gradle:95]: Refactored implementation to return a default value

File: gradle.properties [hunk 5]
```diff
@@ -1,4 +1,4 @@
-VERSION_NAME=6.0.0-SNAPSHOT
+VERSION_NAME=6.0.1-SNAPSHOT
```
[the remark this answers is not in the document]
Comment (hamza-vd) [reply gradle.properties]: Changed to `6.1.0`

File: README.md [hunk 6]
```diff
@@ -307,11 +307,12 @@ By placing a file named `app.properties` in your implementation assets folder (S
...
 | `allow.offline.login.with.invalid.token` | Boolean | false   | Allow offline login when token is no longer valid after a successful login when online and user is forcefully logged out                            |
 | `enable.search.button`                   | Boolean | false   | Enable/Disable search to be triggered only after clicking the search icon in `org.smartregister.view.fragment.BaseRegisterFragment` or its subclass |
-| `feature.profile.images.disabled`            | Boolean | false    | Disable profile image capturing and rendering     |
\ No newline at end of file
+| `feature.profile.images.disabled`            | Boolean | false    | Disable profile image capturing and rendering                                                                                                       |
+| `monitor.height`            | Boolean | false    | Show height unsynced events count in stats view                                                                                                     |
```
Comment (qiarie) [inline README.md:318]: This is already handled in Growth Monitoring. We can remove this entry

File: sample/src/main/assets/app.properties [hunk 7]
```diff
@@ -6,3 +6,4 @@ system.toaster.centered=false
 encrypt.shared.preferences=false
 allow.offline.login.with.invalid.token=false
 ignore.location.deletion = false
+monitor.height=true
```
Comment (qiarie) [inline sample/src/main/assets/app.properties:9]: We can also remove this and use a default value

File: opensrp-core/src/test/java/org/smartregister/util/StatsUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+                ReflectionHelpers.ClassParameter.from(Cursor.class, cursor));
+
+        Map<String, String> syncInfoMap = ReflectionHelpers.getField(statsUtils, "syncInfoMap");
+        Assert.assertEquals("10", syncInfoMap.get(INVALID_EVENTS));
+
+    }
+
+    @After
+    public void destroy() {
+        ReflectionHelpers.setStaticField(CoreLibrary.class, "instance", null);
```
Comment (qiarie) [inline opensrp-core/src/test/java/org/smartregister/util/StatsUtilsTest.java:146]: Let's use `CoreLibrary.destroyInstance()` instead

**Review by qiarie — APPROVED** [review]

**Pull request merged**